.travis.yml

```diff
@@ -70,6 +81,72 @@ _helpers:
   - &install_python pyenv install --skip-existing --keep --verbose "$PYTHON_VERSION"
   - &switch_python pyenv shell "$PYTHON_VERSION"
   - &python_version python --version
+  before_install: &install-from-python_org
+  - |
+    function probe_url() {
```

Comment:
Is it possible to put this complex code into `./tools/` folder and call it from travis config?
It makes a config more understandable.

Reply:
I prefer it here for now because I'd like to see everything in one file. Besides, it's in a separate section anyway. Plus I want to see separate entries in the build log which is impossible with a separate script.

Reply:
This doesn't need to be reused right now. We can talk about it separately but not in this PR. It's easier for me to copy things across projects this way right now.

Reply:
@mjpieters let's talk on Gitter some time: I have a certain vision of where the CI stuff should go in general, architecturally and I want us to be on the same page.